code/__HELPERS/patterns.dm

```diff
@@ -79,3 +79,38 @@
 				turfs_to_check += AdjT
 				results += AdjT
 	return results
+
+///Generates a cone shape. Any other checks should be handled with the resulting list. Can work with up to 359 degrees
+/proc/generate_cone(atom/center, outer_range = 10, inner_range = 1, cone_width = 60, cone_direction = 0, blocked = TRUE)
+	var/right_angle = cone_direction + cone_width/2
+	var/left_angle = cone_direction - cone_width/2
+
+	//These are needed because degrees need to be from 0 to 359 for the checks to function
+	if(right_angle >= 360)
+		right_angle -= 360
+
+	if(left_angle < 0)
+		left_angle += 360
+
+	var/list/turfs_to_check = list(get_turf(center))
```

Comment:
Ca you rename this with rowtocheck  or similar or something for clarity

Reply:
renamed to max_row_count